lms/djangoapps/philu_overrides/tests/test_context_processor.py

```diff
@@ -0,0 +1,590 @@
+"""
+Tests for context processors in philu_overrides
+"""
+
+import mock
+import pytest
+from django.conf import settings
+from django.test import RequestFactory
+from django.urls import reverse
+
+from lms.djangoapps.philu_overrides.constants import (
+    ACTIVATION_ALERT_TYPE,
+    ACTIVATION_ERROR_MSG_FORMAT,
+    ORG_DETAILS_UPDATE_ALERT,
+    ORG_OEF_UPDATE_ALERT
+)
+from lms.djangoapps.philu_overrides.context_processor import (
+    add_nodebb_endpoint,
+    get_cdn_link,
+    get_global_alert_messages
+)
+from student.tests.factories import AnonymousUserFactory, UserFactory
+
+
+@pytest.fixture()
+def request_object(db):  # pylint: disable=unused-argument
+    request = RequestFactory().get('/dummy_url')
+    request.user = UserFactory.create()
+    return request
+
+# Tests for get_global_alert_messages context processor
+
+# ACTIVATION_ERROR_MSG Tests
+
+
+@mock.patch('lms.djangoapps.philu_overrides.context_processor.is_org_detail_platform_overlay_available', mock.Mock())
+@mock.patch('lms.djangoapps.philu_overrides.context_processor.is_org_oef_prompt_available', mock.Mock())
+@mock.patch('lms.djangoapps.philu_overrides.context_processor.get_org_oef_update_prompt', mock.Mock())
+@mock.patch('lms.djangoapps.philu_overrides.context_processor.is_org_detail_prompt_available', mock.Mock())
+@mock.patch('lms.djangoapps.philu_overrides.context_processor.get_org_metric_update_prompt', mock.Mock())
+def test_get_activation_error_alert_message_success(request_object):  # pylint: disable=redefined-outer-name
+    """
+    Test that get_global_alert_messages should return activation error message as the context when:
+        - request is not AJAX,
+        - user is authenticated,
+        - user is inactive,
+        - user is not requesting for activation
+    """
+
+    request = request_object
+    request.user.is_active = False
+
+    expected_global_alert_message = [
+        {
+            'type': ACTIVATION_ALERT_TYPE,
+            'alert': ACTIVATION_ERROR_MSG_FORMAT.format(
+                api_endpoint=reverse('resend_activation_email'),
+                user_id=request.user.id
+            )
+        }
+    ]
+
+    actual_global_alert_message = get_global_alert_messages(request).get('global_alert_messages')
+
+    assert expected_global_alert_message == actual_global_alert_message
```

Comment:
We are doing exactly the same thing in 9 tests, So basically every time we make some changes in request and create an expected response w.r.t to that request. I am not sure if pytests provide some elegant way to do this data driven testing, if it does please use that, otherwise let's please move these two lines in a method and and use that method in all these tests.